### DISCUSSION TITLE
Right setup for CircleCI, running out of memory

### DISCUSSION THREAD

**Discussion by wouter-leistra** [vercel/turborepo#1200] [Help] [answered]
Our product is developed on a monorepo and deployed through circleCI on a per package basis. 35 packages so far. On our circleCI workflows, we use docker with image:node18 in our jobs and run into the issue that a single job like testing all packages when our shared ui component packages is changed using `yarn turbo run --filter=frontend-*` as command gets a error 137 from yarn (out of memory). Even if we set the resource class to xlarge (16gb ram).

Are we doing something wrong? Does anyone have some suggestions on how to avoid running into these out of memory issues?

**Comment by sebiglesias** [comment]
`yarn turbo run --filter=frontend-* --concurrency=1` might be helpful.

https://turborepo.org/docs/reference/command-line-reference#--concurrency

> Defaults to 10. Set/limit the max concurrency of task execution. This must be an integer greater than or equal to 1 or a percentage value like 50%. Use 1 to force serial (i.e. one task at a time) execution. Use 100% to use all available logical processors.

Might be helpful to start with `--concurrency=1` and see if you get an OOM, if you don't try bumping the number until you get an OOM and use that value -1 to save some time.

Hope it helps.

**Comment by izifortune** [comment]
Have the same issue on AWS Codebuild. 16GB of RAM and the container dies as soon as I run turbo commands even with concurrency=1

**Reply by volnei** [reply]
Any sollution here :(?

**Comment by volnei** [comment]
Any solution?

**Comment by anthonyshew** [comment, marked as answer]
Hey, folks, seeing this hasn't seen activity in awhile so I'm going to close it for staleness.

If you're still running into this issue, we'd appreciate you sending us an issue with a reproduction. 🙏
